### PR TITLE
Allow passing run_id to XCom.get_[one|many]

### LINKED ISSUE #16371: Make `XCom.get_one()` + `get_many()` accept run_id
XCom.get_* methods current accept an execution_date, and need to accept a run_id now.

We should add back-compat shims so that any python_operators that call these methods continue to work.

There needs to remain a kwarg called `execution_date` for DAGs/third party code to call, but any uses in Airflow (TI.xcom_pull for instance) should be updated to pass run_id directly)

### PR DESCRIPTION
**Pull request by uranusjr**
Fix #16371

### CODE REVIEW

File: airflow/models/xcom.py [hunk 1]
```diff
@@ -145,13 +150,16 @@ def get_many(
...
+        run_id: Optional[str] = None,
         session: Session = None,
     ) -> Query:
         """
         Composes a query to get one or more values from the xcom table.
 
         :param execution_date: Execution date for the task
         :type execution_date: pendulum.datetime
+        :param run_id: DagRun ID for the task
+        :type run_id: str
```
Comment (mik-laj) [inline airflow/models/xcom.py:162]: Should we execution_date mark as a optional and add explicitly validation for required parameter - run_id or execution_date?

Comment (mik-laj) [reply airflow/models/xcom.py:162]: I think the run_id and execution_date parameters should be mutually exclusive.  WDYT?

**Comment by uranusjr** [comment]
I’ll come back to this after #17719

**Comment by ashb** [comment]
Closed in favour of https://github.com/apache/airflow/pull/18084

**Pull request closed without merging**